xcp_d/utils/concantenation.py

```diff
@@ -221,10 +223,24 @@ def concatenate_nifti(subid, fmridir, outputdir, ses=None, work_dir=None):
     for task in tasklist:
         resbold = natsorted(
             fnmatch.filter(all_func_files,
-                           '*' + task + '*run*_desc-residual*bold*.nii.gz'))
+                           '*_task-' + task + '_*desc-residual*bold*.nii.gz'))
+        resbold_unsmoothed_only = natsorted(
+            fnmatch.filter(all_func_files,
+                           '*_task-' + task + '_*desc-residual_bold*.nii.gz'))
         regressed_dvars = []
         # resbold may be in different space like native space or MNI space or T1w or MNI
-        if len(resbold) > 1:
+        if len(resbold_unsmoothed_only) == 1:
+            res = resbold[0]
+            resid = res.split('task-')[1].partition('_')[-1]
+            # print(resid)
+            for j in datafile:
+                fileid = res.split('task-')[0] + resid.partition('_desc')[0]
+                outfile = fileid + j
+                filex = glob.glob(res.split('task-')[0] + '*task*' + j)
+                if j.endswith('framewisedisplacement_bold.tsv'):
+                    name = f"{fileid}{j.split('.')[0]}-DCAN.hdf5"
+                    make_dcan_df(filex, name)
+        if len(resbold_unsmoothed_only) > 1:
```

Comment:
```suggestion
                    make_dcan_df(filex, name)

        if len(resbold_unsmoothed_only) > 1:
```

Could you include empty lines between if statements, for readability?